### ISSUE TITLE
Crash on Android 12

### ISSUE THREAD

**Issue by igusef** [Kommunicate-io/Kommunicate-Android-Chat-SDK#257]
Hi,
My app targets android 12 and I use the latest version of sdk 2.3.0. Users ran into an issue when push notification comes:

Caused by java.lang.IllegalArgumentException
<my.app.package>: Targeting S+ (version 31 and above) requires that one of FLAG_IMMUTABLE or FLAG_MUTABLE be specified when creating a PendingIntent. Strongly consider using FLAG_IMMUTABLE, only use FLAG_MUTABLE if some functionality depends on the PendingIntent being mutable, e.g. if it needs to be used with inline replies or bubbles.

```
android.app.PendingIntent.checkFlags (PendingIntent.java:382)
android.app.PendingIntent.getActivityAsUser (PendingIntent.java:465)
android.app.PendingIntent.getActivity (PendingIntent.java:451)
android.app.PendingIntent.getActivity (PendingIntent.java:415)
com.applozic.mobicomkit.api.notification.NotificationService.notifyUserForNormalMessage (NotificationService.java:402)
com.applozic.mobicomkit.broadcast.BroadcastService.sendNotificationBroadcast (BroadcastService.java:184)
com.applozic.mobicomkit.api.conversation.MobiComMessageService.sendNotification (MobiComMessageService.java:249)
com.applozic.mobicomkit.api.conversation.MobiComMessageService.addMTMessage (MobiComMessageService.java:230)
com.applozic.mobicomkit.api.conversation.MobiComMessageService.processMessage (MobiComMessageService.java:136)
com.applozic.mobicomkit.api.conversation.MobiComMessageService.syncMessages (MobiComMessageService.java:297)
com.applozic.mobicomkit.api.conversation.ConversationIntentService.onHandleWork (ConversationIntentService.java:84)
androidx.core.app.JobIntentService$CommandProcessor.doInBackground (JobIntentService.java:396)
androidx.core.app.JobIntentService$CommandProcessor.doInBackground (JobIntentService.java:387)
android.os.AsyncTask$3.call (AsyncTask.java:394)
java.util.concurrent.FutureTask.run (FutureTask.java:266)
java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1167)
java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:641)
java.lang.Thread.run (Thread.java:920)
```

**Comment by igusef** [comment]
@rajeev-kommunicate Yes, done.

**Comment by amntoppo** [comment]
Issue has been resolved in SDK version 2.3.5
